pkg/runtime/runtime_test.go

```diff
@@ -1577,6 +1570,13 @@ func TestOnNewPublishedMessageGRPC(t *testing.T) {
 
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
+			port, _ := freeport.GetFreePort()
```

Comment:
Please add a comment about why we create a new runtime every time? This will make sure future authors won't accidentally "improve" by "sharing" this object. Same with other places that you fixed. Thanks